core/src/operations/decorators/mod.rs

```diff
@@ -0,0 +1,53 @@
+mod advice;
+use crate::utils::collections::Vec;
+pub use advice::AdviceInjector;
+use core::fmt;
+
+// DECORATORS
+// ================================================================================================
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Comment:
If in the future we want to have decorators which store strings internally, we'll need to get rid of `Copy`. Maybe it's worth doing it now?

Reply:
Done